### ISSUE TITLE
In Bootsnap 1.12.0 the instance methods in Devise::Models::Recoverable are not loaded (Devise v4.8.1)

### ISSUE THREAD

**Issue by tuomasj** [Shopify/bootsnap#419]
I updated Bootsnap to version 1.12.0 and somehow the instance methods in `Devise::Models::Recoverable` are not available anymore. (and our tests break in 1.12.0)

```
NoMethodError: undefined method `send_reset_password_instructions' for #<User id:1234 ... >
```

I got this error when I ran `rails test` on our application. This was a bit head-scratcher one to submit, I wasn't entirely where I should report this issue, I hope I chose the right issue tracker 🙂

### Steps to reproduce

```
# bootsnap v1.11.1

[6] pry(#<SettingsController>)> Devise::Models::Recoverable.instance_methods
=> [:clear_reset_password_token,
 :clear_reset_password_token?,
 :send_reset_password_instructions,
 :set_reset_password_token,
 :send_reset_password_instructions_notification,
 :reset_password_period_valid?,
 :reset_password]
```

```
# bootsnap 1.12.0

[1] pry(#<SettingsController>)> Devise::Models::Recoverable.instance_methods
=> [:reset_password]
```

### System configuration

Bootsnap 1.12.0
Ruby on Rails 7.0.3
Ruby 3.1.1
Devise 4.8.1

**Comment by casperisfine** [comment]
I'll dig into it, can you try with `1.12.0` these 3 variables one after the other?

  - `DISABLE_BOOTSNAP=1`
  - `DISABLE_BOOTSNAP_LOAD_PATH_CACHE=1`
  - `DISABLE_BOOTSNAP_COMPILE_CACHE=1`

That would help confirm it's caused by bootsnap as well as to pinpoint what part causes this.

**Comment by tuomasj** [comment]
> I'll dig into it, can you try with 1.12.0 these 3 variables one after the other?
>
>    DISABLE_BOOTSNAP=1
>    DISABLE_BOOTSNAP_LOAD_PATH_CACHE=1
>    DISABLE_BOOTSNAP_COMPILE_CACHE=1

I tried all combinations and these had no effect. It still gave the same error.

I'll try the minimal app with `begin/rescue` next

**Comment by tuomasj** [comment]
I tried with minimal app and I could not reproduce it. 

This is totally bizarre bug, everything works as long as Bootsnap is in version 1.11.1, once I bump it to 1.12.0, at least `Devise::Models::Recoverable` is not loaded. I've verified it with putting a `raise` in to the beginning of that file.

`DISABLE_BOOTSNAP=1` has no effect either.

Since I cannot reproduce this in any other way, let's close this issue.
